ui/src/data_explorer/components/Table.js

```diff
@@ -31,50 +35,74 @@ const ChronoTable = React.createClass({
     query: shape({
       host: arrayOf(string.isRequired).isRequired,
       text: string.isRequired,
-    }),
+    }).isRequired,
     containerWidth: number.isRequired,
     height: number,
+    onEditRawStatus: func,
   },
 
   getInitialState() {
     return {
-      cellData: {
-        columns: [],
-        values: [],
-      },
+      cellData: emptyCells,
       columnWidths: {},
     }
   },
 
   getDefaultProps() {
     return {
-      height: 600,
+      height: 1000,
     }
   },
 
-  fetchCellData(query) {
+  componentDidMount() {
+    this.fetchCellData(this.props.query)
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.query.text === nextProps.query.text) {
+      return
+    }
+
+    this.fetchCellData(nextProps.query)
+  },
+
+
+  async fetchCellData(query) {
+    if (!query || !query.text) {
```

Comment:
I thought JS `||` short-circuits if the left side is true? So
`if !query` or
`if query || !query.text`